src/coredump/coredump.c

```diff
@@ -1676,7 +1676,7 @@ static int gather_pid_mount_tree_fd(const Context *context) {
                            "(sd-mount-tree)",
                            /* except_fds= */ NULL,
                            /* n_except_fds= */ 0,
-                           FORK_RESET_SIGNALS|FORK_DEATHSIG_SIGKILL|FORK_LOG,
+                           FORK_RESET_SIGNALS|FORK_DEATHSIG_SIGKILL|FORK_LOG|FORK_WAIT,
```

Comment:
afaics you don't need the `child` parameter anymore now. hence you can set that to NULL and get rid of the local var for it.